### PR TITLE
Enhancement of JDBC adpater (to support pgvector recall testing)

### PR DESCRIPTION
**Pull request by yabinmeng**
- Add "use_hikaricp" option: enable/disable using Hikari connection pooling. 
- Various code cleanup based on Shooky's comments
- Update the pgvector testing scenario yaml file

### CODE REVIEW

File: adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java [hunk 1]
```diff
@@ -34,18 +34,49 @@ public abstract class JDBCOp implements CycleOp {
...
+
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("A new JDBC connection ({}) is successfully created: {}",
+                        connectionName, connection);
+                }
+
+                return  connection;
+            }
+            catch (Exception ex) {
+                String exp = "Exception occurred while attempting to create a connection using the HikariDataSource";
```
Comment (msmygit) [inline adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java:76]:
```suggestion
                String exp = "Exception occurred while attempting to create a connection";
```

File: adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java [hunk 2]
```diff
@@ -34,18 +34,49 @@ public abstract class JDBCOp implements CycleOp {
...
+
+                // Register 'vector' type
+                JDBCPgVector.addVectorType(connection);
+
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("A new JDBC connection ({}) is successfully created: {}",
+                        connectionName, connection);
+                }
+
+                return  connection;
```
Comment (msmygit) [inline adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java:73]:
```suggestion
                return connection;
```

File: adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java [hunk 3]
```diff
@@ -48,208 +43,170 @@ public class JDBCSpace implements AutoCloseable {
...
-
         try {
-            waitUntilAllOpFinished(System.currentTimeMillis());
-
+            logger.info("Shutting down JDBCSpace -- total {} of connections is being closed ...", connections.size());
             for (Connection connection : connections.values()) {
-                connection.close();
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Close connection : {}", connection);
+                    connection.close();
```
Comment (msmygit) [inline adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java:200]: This closing logic will not be applied if the logger doesn't have debug enabled

File: adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java (hunk 1 above)
Comment (yabinmeng) [reply adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java]: fixed

File: adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java (hunk 3 above)
Comment (yabinmeng) [reply adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java]: Thanks. Fixed

File: adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java (hunk 2 above)
Comment (yabinmeng) [reply adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCOp.java]: Fixed

**Pull request merged**